js/SearchResult.js

```diff
@@ -0,0 +1,105 @@
+/* search result class */
+class SearchResult {
+  /* search result class constructor */
+  constructor(resultElement) {
+    this.resultList = resultElement;
+  }
+
+  highlightText(company, searchValue, searchKey, elementToAppend) {
+    let highLightArea = [0, 0];
+
+    const highLightSpanLink = document.createElement("span");
+    const regSpanLink = document.createElement("span");
+    highLightSpanLink.classList.add("highlight");
+    /* search value to upper char and slicing */
+    highLightSpanLink.className = "highlight";
+    searchValue = searchValue.charAt(0).toUpperCase() + searchValue.slice(1);
+    /* if equals symbol */
+    highLightArea[0] =
+      searchKey === "symbol"
+        ? company[searchKey].indexOf(searchValue[0].toUpperCase())
+        : company[searchKey].indexOf(searchValue);
+    /* if exists */
+    if (highLightArea[0] !== -1) {
+      highLightArea[1] = highLightArea[0] + searchValue.length;
+      /* creating sub string */
+      highLightSpanLink.innerText = company[searchKey].substring(
+        highLightArea[0],
+        highLightArea[1]
+      );
+      /* returning new string with replacement "" */
+      regSpanLink.innerText = company[searchKey].replace(
+        highLightSpanLink.innerText,
+        ""
+      );
+      if (searchKey === "symbol") {
+        regSpanLink.innerText += ")";
+        highLightSpanLink.innerText = "(" + highLightSpanLink.innerText;
+      }
+      elementToAppend.append(highLightSpanLink, regSpanLink);
+    } else {
+      regSpanLink.innerText = company[searchKey];
+      if (searchKey === "symbol") {
+        regSpanLink.innerText = "(" + regSpanLink.innerText + ")";
+      }
+      elementToAppend.append(regSpanLink);
+    }
+  }
+
+  /* rendering the results */
+  renderResults({ companies, searchValue }) {
+    const fragment = new DocumentFragment();
```

Comment:
I'm a big fan of DocumentFragment, if your are intersted in it, read this:
https://www.webtips.dev/document-fragments